### PR TITLE
feat: add .env inheritance support in Envied and EnviedGenerator

### LINKED ISSUE #189: Support environment inheritance for .env files
**Description:**  
Currently, each environment file (e.g., `.env.dev`, `.env.prod`) must redefine all variables, even when many share the same default values. It would be useful to support environment inheritance, allowing a base file (e.g., `.env.defaults`) to define defaults that other environment files can extend.  

**Example use case:**  
- `.env.defaults` contains shared defaults.  
- `.env.dev` and `.env.prod` inherit from `.env.defaults`.  
- These environment-specific files only override values or add new entries.  

**Benefits:**  
- Reduces duplication across environment files.  
- Simplifies maintenance when updating shared variables.  
- Makes configuration more consistent and easier to manage across multiple environments.

### PR DESCRIPTION
**Pull request by techouse**
This pull request introduces support for environment file inheritance in the Envied package, allowing users to specify multiple `.env` files whose values are loaded and merged in order. The changes include updates to the `Envied` annotation, enhancements to the environment loading logic, and comprehensive tests and documentation to cover the new feature.

**Environment inheritance feature:**

- Added a new `inheritFrom` parameter to the `Envied` annotation, allowing users to specify an ordered list of `.env` files to be loaded before the primary file, with later files overriding earlier values. (`packages/envied/lib/src/envied_base.dart`, `packages/envied_generator/lib/src/generator.dart`) [[1]](diffhunk://#diff-2840575a98cec94a77ff8dd956d3a05f02b8e0881ad337cd0f73258f0621391dR12-R17) [[2]](diffhunk://#diff-2840575a98cec94a77ff8dd956d3a05f02b8e0881ad337cd0f73258f0621391dR109) [[3]](diffhunk://#diff-70739d136f7c66633d926b87532dbcad7e1ba0cf033df53d47945e416b06d8caL42-R51) [[4]](diffhunk://#diff-70739d136f7c66633d926b87532dbcad7e1ba0cf033df53d47945e416b06d8caL69-R68) [[5]](diffhunk://#diff-70739d136f7c66633d926b87532dbcad7e1ba0cf033df53d47945e416b06d8caR78-R100) [[6]](diffhunk://#diff-70739d136f7c66633d926b87532dbcad7e1ba0cf033df53d47945e416b06d8caL109-R123) [[7]](diffhunk://#diff-70739d136f7c66633d926b87532dbcad7e1ba0cf033df53d47945e416b06d8caR154-R173)

- Updated the environment variable loading logic to support merging values from multiple files, preserving the correct override order and supporting variable interpolation across inherited files. (`packages/envied_generator/lib/src/load_envs.dart`, `packages/envied_generator/lib/src/parser.dart`) [[1]](diffhunk://#diff-4082e34afa3805d610580778a46356bf2acca77e053c28db84f7a25ed801d57bL11-R35) [[2]](diffhunk://#diff-4082e34afa3805d610580778a46356bf2acca77e053c28db84f7a25ed801d57bL24-R45) [[3]](diffhunk://#diff-f323253b8575b959a00eb4e09fa11d80d587925e0a203ef0033a7ec8d5188ac7L22-R37)

**Documentation improvements:**

- Extended the README with a new section and example for environment inheritance, explaining usage and override behavior. (`packages/envied/README.md`) [[1]](diffhunk://#diff-4a011b1a508b3a235167e0553e65fc6aa465c07b699440bb2b0113a83906991fR39) [[2]](diffhunk://#diff-4a011b1a508b3a235167e0553e65fc6aa465c07b699440bb2b0113a83906991fR296-R330)

**Testing and validation:**

- Added new and updated tests to verify the behavior of the `inheritFrom` parameter, including edge cases with missing files and multiple inheritance scenarios. (`packages/envied/test/envy_test.dart`, `packages/envied_generator/test/envy_generator_test.dart`, `packages/envied_generator/test/src/generator_tests_with_inheritance.dart`, and new test `.env` files) [[1]](diffhunk://#diff-23e417b942877f8a1fe0b30292720477c09fd7cdf3a37e27eb101d7b4cb56a8bR9) [[2]](diffhunk://#diff-23e417b942877f8a1fe0b30292720477c09fd7cdf3a37e27eb101d7b4cb56a8bR20-R29) [[3]](diffhunk://#diff-ca5077f67806371f7ea35bda179e4b5e294aa6c1f0b7ddcd6bbfbf3bc0b385c5R31-R38) [[4]](diffhunk://#diff-c4c02dccf70f6fc863c51214d90093f80679c858f8cb61c7c4f880ef850e41f9R1-R112) [[5]](diffhunk://#diff-e69fde0da7076745a4a166c313aa97346ba71de48ebc705253301e241471a380R1-R2) [[6]](diffhunk://#diff-ae181e97425b40bb03c38346c928de989b33d4b532bc44d97d9e47ee8a68df5aR1-R2) [[7]](diffhunk://#diff-18dbab966770046aeddf8615c402360d2f131dffc661741a8c5300284d412c03R1-R7) [[8]](diffhunk://#diff-87492f439436b181c7e9c168668f85949efe58f41462c8d61e8089efa03b134cR1-R3) [[9]](diffhunk://#diff-d3dd39e98e7d3042aa5ffd45d9dde766fbcaf02eb0b7055d9f9ff53cbbb8759aR1-R2)

**Minor improvements:**

- Improved handling of empty `path` values in the `Envied` annotation to default to `.env`. (`packages/envied/lib/src/envied_base.dart`, `packages/envied_generator/lib/src/generator.dart`, `packages/envied_generator/test/src/generator_tests.dart`) [[1]](diffhunk://#diff-2840575a98cec94a77ff8dd956d3a05f02b8e0881ad337cd0f73258f0621391dL112-R119) [[2]](diffhunk://#diff-9164cd585484b603acaa9d4a9ff274c15b2b83134c221af70ab9e008c372de61R23-R31)

**Changelog update:**

- Updated the changelog to reflect the addition of the environment inheritance feature. (`CHANGELOG.md`)

### CODE REVIEW

File: packages/envied/lib/src/envied_base.dart [hunk 1]
```diff
@@ -109,7 +116,7 @@ final class Envied {
     this.interpolate = true,
     this.rawStrings = false,
     this.randomSeed,
-  }) : path = path ?? '.env',
+  }) : path = path == null || path == '' ? '.env' : path,
```
[the remark this answers is not in the document]
Comment (techouse) [reply packages/envied/lib/src/envied_base.dart:119]: That won't work because it's an "Invalid constant value"

File: packages/envied_generator/pubspec.yaml [hunk 2]
```diff
@@ -12,7 +12,7 @@ dependencies:
   analyzer: ">=8.0.0 <14.0.0"
   build: ^4.0.3
   code_builder: ^4.11.0
-  envied: ^1.3.3
+  envied: ^1.3.3  # TODO(@techouse) make sure to update to the very latest before release
```
[the remark this answers is not in the document]
Comment (techouse) [reply packages/envied_generator/pubspec.yaml:15]: Yes, I'll do this when cutting a new release.

**Comment by techouse** [comment]
@petercinibulk this should be ready for review now :)
